test/OrchardCore.Tests/Modules/OrchardCore.Tenants/Controllers/AdminControllerTests.cs

```diff
@@ -0,0 +1,247 @@
+                .Returns(Task.FromResult((IDictionary<string, FeatureProfile>)new Dictionary<string, FeatureProfile>
+                {
+                    { "Feature Profile", new FeatureProfile() }
+                }));
+        }
+
+        [Fact]
+        public async Task CreateTenant()
+        {
+            // Arrange
+            var tenants = new List<(string Name, string UrlPrefix, string Hostname, bool IsValid)>
+            {
+                ("Tenant1", "tenant1", "example1.com", true),
+                ("Tenant2", "", "example1.com,example2.com", true),
+                ("Tenant3", "tenant1", "example1.com", false),
+                ("Tenant4", "", "example1.com", false),
+                ("Tenant5", "", "example2.com", false),
+                ("Tenant6", "tenant1", "example2.com", true),
+                ("Tenant7", "tenant2", "example2.com", true),
+                ("Tenant8", "tenant1", "example3.com", true),
+                ("Tenant9", "tenant2", "example3.com", true),
+                ("Tenant10", null, "example4.com", true)
+            };
+
+            // Act & Assert
+            foreach (var (Name, UrlPrefix, Hostname, IsValid) in tenants)
+            {
+                var controller = CreateController();
+                var viewModel = new EditTenantViewModel
+                {
+                    Name = Name,
+                    RequestUrlPrefix = UrlPrefix,
+                    RequestUrlHost = Hostname,
+                    FeatureProfile = "Feature Profile"
+                };
+
+                await controller.Create(viewModel);
+
+                Assert.True(IsValid == controller.ModelState.IsValid);
+            }
+        }
+
+        [Fact]
+        public async Task CreateTenantShouldChecksHostnameIfItAlreadyUsedInMultipleHostnames()
+        {
+            // Arrange & Act & Assert
+            var controller = CreateController();
+            var viewModel = new EditTenantViewModel
+            {
+                Name = "Tenant11",
+                RequestUrlPrefix = String.Empty,
+                RequestUrlHost = "example5.com, example6.com",
+                FeatureProfile = "Feature Profile"
+            };
+
+            await controller.Create(viewModel);
+
+            Assert.True(controller.ModelState.IsValid);
+
+            controller = CreateController();
+
+            viewModel = new EditTenantViewModel
+            {
+                Name = "Tenant12",
+                RequestUrlPrefix = String.Empty,
+                RequestUrlHost = "example6.com",
+                FeatureProfile = "Feature Profile"
+            };
+
+            await controller.Create(viewModel);
+
+            Assert.False(controller.ModelState.IsValid);
+            Assert.Equal("A tenant with the same host and prefix already exists.", controller.ModelState.First().Value.Errors.First().ErrorMessage);
+        }
+
+        private AdminController CreateController()
```

Comment:
As I mentioned before, you don't need all this ceremony if you're testing the service, what's the point here, not the controller.

Reply:
After we introduced a service, no need to check the controller